scripts/pre_commit_linter.py

```diff
@@ -1312,6 +1335,89 @@ def _check_js_component_name_and_count(self):
             print ''
             return summary_messages
 
+    def _check_mandatory_patterns(self):
+        """This function checks that all files contain the mandatory
+        patterns.
+        """
+        if self.verbose_mode_enabled:
+            print 'Starting mandatory patterns check'
+            print '----------------------------------------'
+
+        summary_messages = []
+        failed = False
+
+        with _redirect_stdout(_TARGET_STDOUT):
+            for filepath in self.all_filepaths:
+                # This boolean list keeps track of the regex matches
+                # found in the file.
+                pattern_found_list = []
+                file_content = FileCache.readlines(filepath)
+                for index, regexp_to_check in enumerate(
+                        MANDATORY_PATTERNS_REGEXP):
+                    if (any([filepath.endswith(
+                            allowed_type) for allowed_type in (
+                                regexp_to_check['included_types'])]) and (
+                                    not any([
+                                        filepath.endswith(
+                                            pattern) for pattern in (
+                                                regexp_to_check[
+                                                    'excluded_files'] +
+                                                regexp_to_check[
+                                                    'excluded_dirs'])]))):
+                        pattern_found_list.append(False)
+                        for line in file_content:
+                            if re.search(regexp_to_check['regexp'], line):
+                                pattern_found_list[index] = True
+                                break
+                if not all(pattern_found_list):
+                    failed = True
+                    for index, pattern_found in enumerate(
+                            pattern_found_list):
+                        if not pattern_found:
+                            print '%s --> %s' % (
+                                filepath,
+                                MANDATORY_PATTERNS_REGEXP[index]['message'])
+
+                if filepath.endswith('.js'):
+                    pattern_found_list = []
```

Comment:
Can we pull out lines 1382-1405 in a separate function since we're basically duplicating them above? Also, when in the future, we try to extend this check, we will need these lines again. Therefore, it might be better to pull it off in a separate function. Let me know if there are any doubts. Thanks!

Reply:
Done!